File: go-client-sdk/pkg/models/operations/getbatchcrmcontact.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetBatchCrmContactQueryParams struct {
	AccessToken string  `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool   `queryParam:"style=form,explode=true,name=allFields"`
	Ids         *string `queryParam:"style=form,explode=true,name=ids"`
}

type GetBatchCrmContactResponseBody struct {
	Contacts   []shared.Contact `json:"contacts,omitempty"`
	InvalidIds []string         `json:"invalidIds,omitempty"`
}

type GetBatchCrmContactRequest struct {
	QueryParams GetBatchCrmContactQueryParams
}

type GetBatchCrmContactResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetBatchCrmContactResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getbatchcrmcontact.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetBatchCrmContactQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=ids" })
  ids?: string;
}


export class GetBatchCrmContactResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=contacts", elemType: shared.Contact })
  contacts?: shared.Contact[];

  @SpeakeasyMetadata({ data: "json, name=invalidIds" })
  invalidIds?: string[];
}


export class GetBatchCrmContactRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetBatchCrmContactQueryParams;
}


export class GetBatchCrmContactResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetBatchCrmContactResponseBody;
}
